File: CodeChef/OCT221D/BUILDINGRACE.py
for i in range(int(input())):
    a, b, x, y = map(int, input().split(" "))
    speed_a = 60 / x
    speed_b = 60 / y
    time_a = a*speed_a
    time_b = b*speed_b
    if time_a > time_b:
        print("Chefina")
    elif time_b > time_a:
        print("Chef")
    else:
        print("Both")

File: CodeChef/OCT221D/BUILDINGRACE.cpp
#include <iostream>
using namespace std;

int main()
{
    int T;
    cin >> T;
    for (int i = 0; i < T; i++)
    {
        float A, B, X, Y;
        cin >> A >> B >> X >> Y;
        float c, d;
        c = 1 / X;
        d = 1 / Y;
        c = c * A;
        d = d * B;
        if (c > d)
        {
            cout << "Chefina" << endl;
        }
        else if (c < d)
        {
            cout << "Chef" << endl;
        }
        else
        {
            cout << "Both" << endl;
        }
    }

    return 0;
}